addons/point_of_sale/static/src/app/screens/partner_list/partner_list.js

```diff
@@ -55,15 +55,85 @@ export class PartnerList extends Component {
             this.notification.add(_t('No more customer found for "%s".', this.state.query), 3000);
         }
     }
+
+    goToOrders() {
+        this.back(true);
+        const partner = this.state.editModeProps.partner;
+        const partnerHasActiveOrders = this.pos
+            .get_order_list()
+            .some((order) => order.partner?.id === partner.id);
+        const ui = {
+            searchDetails: {
+                fieldName: "PARTNER",
+                searchTerm: partner.name,
+            },
+            filter: partnerHasActiveOrders ? "" : "SYNCED",
+        };
+        this.pos.showScreen("TicketScreen", { ui });
+    }
+
+    confirm() {
+        this.props.resolve({ confirmed: true, payload: this.state.selectedPartner });
+        this.pos.closeTempScreen();
+    }
+    activateEditMode() {
+        this.state.detailIsShown = true;
+    }
+    // Getters
+
+    get_partners_sorted(max_count) {
+        const partners = [];
+        const resPartner = this.pos.models["res.partner"].getAll();
+        max_count = max_count ? Math.min(resPartner.length, max_count) : resPartner.length;
+
+        for (var i = 0; i < max_count; i++) {
+            partners.push(this.pos.models["res.partner"].get(resPartner[i].id));
+        }
+
+        return partners;
+    }
+
+    partner_search_string(partner) {
+        var str = partner.name || "";
+        if (partner.barcode) {
+            str += "|" + partner.barcode;
+        }
+        if (partner.phone) {
+            str += "|" + partner.phone.split(" ").join("");
+        }
+        if (partner.mobile) {
+            str += "|" + partner.mobile.split(" ").join("");
+        }
+        if (partner.email) {
+            str += "|" + partner.email;
+        }
+        if (partner.vat) {
+            str += "|" + partner.vat;
+        }
+        if (partner.parent_name) {
+            str += "|" + partner.parent_name;
+        }
+        str = "" + partner.id + ":" + str.replace(":", "").replace(/\n/g, " ") + "\n";
+        return str;
+    }
+
+    get_partners_searched() {
+        return fuzzyLookup(
```

Comment:
this will be very slow. I remember Sam optimized this already, can't we keep his implementation?

Reply:
I use the same on for product now, I tested with 6000+ products and it run like a charm, what do you think ?

Reply:
You should test it with 20k partners. I think that's the maximum number of `res.partner`s that we load.

Reply:
I will test